File: tutorial/04-recursive/problems/07-tower-of-hanoi.java
class SoftP0407{
    public static int solution(int n ,char from, char aux, char to){
        if(n==0)return 0;
        int res1= SoftP0407.solution(n-1, from, to, aux);
        int res2=SoftP0407.solution(n-1, aux, from, to);
        return res1+res2+1;
    }
}

File: tutorial/04-recursive/problems/07-tower-of-hanoi.py
class SoftP0407:
    def solution(self,n,fromm,aux,to):
        if(n==0):
            return 0
        res1= self.solution(n-1,fromm,to,aux)
        res2= self.solution(n-1,aux,fromm,to)
        return res1+res2+1